Source/Utils/WarpXMovingWindow.cpp

```diff
@@ -57,6 +57,199 @@
 
 using namespace amrex;
 
+namespace
+{
+
+    /** This function shifts a MultiFab in a given direction
+    *
+    * \param[in,out] mf the MultiFab to be shifted
+    * \param[in] geom the Geometry object associated to the level of the MultiFab mf
+    * \param[in] num_shift magnitude of the shift (cell number)
+    * \param[in] dir direction of the shift
+    * \param[in] safe_guard_cells flag to enable "safe mode" data exchanges with more guard cells
+    * \param[in] do_single_precision_comms flag to enable single precision communications
+    * \param[in,out] cost the pointer to the data structure holding costs for timer-based load-balance
+    * \param[in] external_field the external field (used to initialize EM fields)
+    * \param[in] useparser flag to enable the use of a field parser to initialize EM fields
+    * \param[in] field_parser the field parser
+    * \param[in] PMLRZ_flag flag to enable a special treatment for PML in RZ simulations
+    */
+    void shiftMF (
+        amrex::MultiFab& mf, const amrex::Geometry& geom,
+        int num_shift, int dir,
+        bool safe_guard_cells, bool do_single_precision_comms,
+        amrex::LayoutData<amrex::Real>* cost,
+        amrex::Real external_field=0.0, bool useparser = false,
+        amrex::ParserExecutor<3> const& field_parser={},
+        const bool PMLRZ_flag = false)
+    {
+        using namespace amrex::literals;
+        WARPX_PROFILE("::shiftMF()");
+        const amrex::BoxArray& ba = mf.boxArray();
+        const amrex::DistributionMapping& dm = mf.DistributionMap();
+        const int nc = mf.nComp();
+        const amrex::IntVect& ng = mf.nGrowVect();
+
+        AMREX_ALWAYS_ASSERT(ng[dir] >= std::abs(num_shift));
```

Comment:
Previously, this was `AMREX_ALWAYS_ASSERT(ng[dir] >= num_shift);`
however, it seems to me that, since `num_shift` can be negative, we need an absolute value here. What do you think @ax3l ?

Reply:
Wait, we can shift backwards (or just know it is technically possible because `num_shift` is an `int`)? :D 

Anyway, LGTM ;)